internal/services/codesigning/trusted_signing_account_resource.go

```diff
@@ -0,0 +1,221 @@
+		Func: func(ctx context.Context, meta sdk.ResourceMetaData) error {
+			client := meta.Client.CodeSigning.Client.CodeSigningAccounts
+
+			var model TruestedSigningAccountModel
+			if err := meta.Decode(&model); err != nil {
+				return err
+			}
+
+			subscriptionID := meta.Client.Account.SubscriptionId
+			id := codesigningaccounts.NewCodeSigningAccountID(subscriptionID, model.ResourceGroupName, model.Name)
+			existing, err := client.Get(ctx, id)
+			if !response.WasNotFound(existing.HttpResponse) {
+				if err != nil {
+					return fmt.Errorf("retrieving %s: %v", id, err)
+				}
+				return meta.ResourceRequiresImport(m.ResourceType(), id)
+			}
+
+			req := codesigningaccounts.CodeSigningAccount{}
+			req.Name = &model.Name
+			req.Location = model.Location
+			req.Properties = pointer.To(codesigningaccounts.CodeSigningAccountProperties{})
+			req.Properties.Sku = pointer.To(codesigningaccounts.AccountSku{
+				Name: codesigningaccounts.SkuName(model.SkuName),
+			})
+			req.Tags = &model.Tags
+
+			err = client.CreateThenPoll(ctx, id, req)
+			if err != nil {
+				return fmt.Errorf("creating %s: %v", id, err)
+			}
+
+			meta.SetID(id)
+			return nil
+		},
+	}
+}
+
+func (m TrustedSigningAccountResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, meta sdk.ResourceMetaData) error {
+			id, err := codesigningaccounts.ParseCodeSigningAccountID(meta.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			client := meta.Client.CodeSigning.Client.CodeSigningAccounts
+			result, err := client.Get(ctx, *id)
+			if err != nil {
+				return err
+			}
+
+			if result.Model == nil {
+				return fmt.Errorf("retrieving %s got nil model", id)
+			}
+			var output TruestedSigningAccountModel
+			output.Name = id.CodeSigningAccountName
+			output.ResourceGroupName = id.ResourceGroupName
+
+			if result.Model == nil {
+				return fmt.Errorf("Get response nil Model")
+			}
+			model := result.Model
+			output.Location = model.Location
+			output.Tags = pointer.From(model.Tags)
+			if ptrProp := model.Properties; ptrProp != nil {
+				itemProp := *ptrProp
+				output.AccountUri = pointer.From(itemProp.AccountUri)
+				if ptrSku := itemProp.Sku; ptrSku != nil {
+					itemSku := *ptrSku
+					output.SkuName = string(itemSku.Name)
+				}
+			}
+
+			return meta.Encode(&output)
+		},
+	}
+}
+
+func (m TrustedSigningAccountResource) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 10 * time.Minute,
+		Func: func(ctx context.Context, meta sdk.ResourceMetaData) (err error) {
+			client := meta.Client.CodeSigning.Client.CodeSigningAccounts
+			id, err := codesigningaccounts.ParseCodeSigningAccountID(meta.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+			var model TruestedSigningAccountModel
+			if err = meta.Decode(&model); err != nil {
+				return fmt.Errorf("decoding %s: %+v", id, err)
+			}
+
+			var upd codesigningaccounts.CodeSigningAccountPatch
```

Comment:
Minor but `upd` is not a known acronym or shorthand for `update` which isn't a long variable name in itself. Can we please be consistent with other resources in the provider and name these one of `payload`, `request`, `param` or spell this out completely to `update`?